src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimal() {
+    for (let i = 0; i < Animal.alive.length; i++) {
+      if (Animal.alive[i].health <= 0) {
+        Animal.alive.splice(i, 1);
+        i--;
+      }
+    }
```

Comment:
The `deleteAnimal` method uses a nested loop which can be inefficient. Consider using `filter` to create a new array excluding the dead animals, and then reassigning it to `Animal.alive`.